src/assets/Keycloak/keycloak.md

```diff
@@ -0,0 +1,69 @@
+---
+
+## Keycloak
+[Keycloak](https://github.com/keycloak/keycloak) is an open-source project for Red Hat SSO. It provides an `Identity and Access Management` solutions designed to secure application services.
+Additionally, it enables users to authenticate through various identity providers and use fine-grained permissions for regulating access to Software as a Service (SaaS) applications.
+Keycloak alternatives are Auth0, Zluri, Microsoft Azure Active Directory, Okta or WSO2 Identity Server.
+
+It facilitates the creation of a user database with customizable roles and groups, offering functionalities such as user management, registration, and password policy enforcement.
+This makes it a comprehensive Identity and Access Management (IAM) solution for contemporary applications. Keycloak's API enables the integration of all these features into your application
+without the need for additional coding.
+
+## Keycloak Features
+![Keycloak features](/src/assets/Keycloak/images/keycloak-features.png)
+
+1. `Identity Broker:`
+   Identity Broker acts as a middleman, linking various service providers with different identity providers. It establishes trust with external identity providers to utilize their identities for accessing internal services offered by service providers.
+   Imagine a textile company, which has two Keycloak setups: one for customer and another for internal employees. If you want your employees to access external services without creating new accounts on customer Keycloak, you can use the customer Keycloak
+   as the mediator and the customer Keycloak as the source in this scenario.
+
+2. `User Federation:`
+   Keycloak provides the ability to integrate with existing LDAP or Active Directory servers. It also supports implementation of your own provider such as a relational database.
+
+3. `Standard Protocols:`
+   Keycloak is based on three standard protocols i.e., OpenID Connect, OAuth2.0, and SAML.
+
+4. `Password Policies:`
+   Keycloak contain different password policies i.e., HashAlgorithm, Hashing Iterations, Digits, Lowercase Characters, Regular Expression and so on.
+
+5. `Single-Sign On:`
+   Keycloak facilitates a seamless login experience where users only need to sign in once to access multiple applications, eliminating the need for repeated logins.
+   This not only streamlines user experience but also simplifies identity management for administrators.
+
+6. `Social Login:`
+   Keycloak has built-in support for Google, GitHub, and Facebook which help us to use social identity providers.
+
+## Keycloak Drawbacks
+Despite its extensive features, Keycloak does have certain limitations. One of these is the need for a more varied implementation approach.
+Additionally, the following are some of the drawbacks associated with Keycloak:
+1. `Complex Server Deployment:` The manual process involved in deploying Keycloak on a server can be complicated, potentially affecting overall productivity.
```

Comment:
I wouldn't use `xxx` syntax to highlight your points since this is for keywords/code snippets in plain text. I would rather make it bold or use the <strong> HTML tag for highlighting your points.